src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -747,10 +747,41 @@ private void GetSpillSizeAndOffsetAtIndex(int index, out int size, out int offse
             offset = 0;
             for (int i = 0; i < index; i++)
             {
-                offset += _spilledExpressions[i].Type.GetElementSize().AsInt;
+                offset = PadNextOffset(_spilledExpressions[i].Type, offset);
             }
         }
 
+        public int PadNextOffset(TypeDesc type, int atOffset)
+        {
+            var size = type is DefType && type.IsValueType ? ((DefType)type).InstanceFieldSize : type.Context.Target.LayoutPointerSize;
```

Comment:
I believe GetElementSize has this logic already.

Reply:
any idea why `MetadataFieldLayoutAlgorithm` does it this way?

Reply:
No idea. It might just be historical.